src/otx/core/model/module/detection.py

```diff
@@ -32,32 +35,25 @@ def __init__(
         torch_compile: bool,
         optimizer: list[OptimizerCallable] | OptimizerCallable = lambda p: torch.optim.SGD(p, lr=0.01),
         scheduler: list[LRSchedulerCallable] | LRSchedulerCallable = torch.optim.lr_scheduler.ConstantLR,
+        metric: MetricCallable | None = None,
     ):
         super().__init__(
             otx_model=otx_model,
             torch_compile=torch_compile,
             optimizer=optimizer,
             scheduler=scheduler,
+            metric=metric,
         )
 
-        self.val_metric = MeanAveragePrecision()
-        self.test_metric = MeanAveragePrecision()
-
-    def on_validation_epoch_start(self) -> None:
-        """Callback triggered when the validation epoch starts."""
-        self.val_metric.reset()
-
-    def on_test_epoch_start(self) -> None:
-        """Callback triggered when the test epoch starts."""
-        self.test_metric.reset()
+        param_dict = {}
+        if metric:
+            sig = inspect.signature(metric)
+            for name, param in sig.parameters.items():
+                param_dict[name] = param.default
+            param_dict.pop("kwargs")
+            metric = metric(**param_dict)  # type: ignore[call-arg]
```

Comment:
What's the reason that this is created at `on_test_epoch_start()`, not its `__init__()`?